### ISSUE TITLE
Serial vs. Parallel output of a number of values

### ISSUE THREAD

**Issue by andrewdavidmackenzie** [andrewdavidmackenzie/flow#148]
When a function can output a number of values, have a way to allow it to output them in series, or in parallel?

e.g. a range of numbers....
Series: one is output at a time, on each invocation.

Parallel: output a range in one go and the system (maybe) divides them between multiple receivers. If receivers receive a range of numbers one by one will depend on their definition, but a range would be a different data type (array of numbers).

Maybe, if a function outputs an array of something, be able to specify next function as a serial processor, or a "fan out" that would split it into *many* individual numbers that maybe spawns many runnables in parallel (depends on the runtime how many, from 1 up).
